Bugsnag/Breadcrumbs/BugsnagBreadcrumbs.h

```diff
@@ -25,30 +27,31 @@ typedef void (^BSGBreadcrumbConfiguration)(BugsnagBreadcrumb *_Nonnull);
 /**
  * Store a new breadcrumb with a provided message.
  */
-- (void)addBreadcrumb:(NSString *_Nonnull)breadcrumbMessage;
+- (void)addBreadcrumb:(NSString *)breadcrumbMessage;
 
 /**
  *  Store a new breadcrumb configured via block.
  *
  *  @param block configuration block
  */
-- (void)addBreadcrumbWithBlock:
-    (void (^_Nonnull)(BugsnagBreadcrumb *_Nonnull))block;
-
-/**
- * Generates an array of dictionaries representing the current buffer of breadcrumbs.
- */
-- (NSArray *_Nonnull)arrayValue;
+- (void)addBreadcrumbWithBlock:(BSGBreadcrumbConfiguration)block;
 
 /**
  * Returns an array containing the current buffer of breadcrumbs.
  */
-- (NSArray<BugsnagBreadcrumb *> *_Nonnull)getBreadcrumbs;
+- (NSArray<BugsnagBreadcrumb *> *)getBreadcrumbs;
 
 /**
- * The types of breadcrumbs which will be automatically captured.
- * By default, this is all types.
+ * Returns the breadcrumb JSON dictionaries stored on disk.
  */
-@property BSGEnabledBreadcrumbType enabledBreadcrumbTypes;
+- (nullable NSArray<NSDictionary *> *)cachedBreadcrumbs;
+
+#pragma mark - Private
+
+@property (nonatomic, readonly, strong) NSMutableArray<BugsnagBreadcrumb *> *breadcrumbs;
+
+@property (nonatomic, readonly, strong) dispatch_queue_t readWriteQueue;
```

Comment:
Is this an organizational preference or does some other object/function else need to know about the queue?

Reply:
The unit tests need access to it - but don't worry it will be removed in the next PR!